src/server/typingsInstaller/typingsInstaller.ts

```diff
@@ -64,6 +64,20 @@ namespace ts.server.typingsInstaller {
         onRequestCompleted: RequestCompletedAction;
     }
 
+    function isPackageOrBowerJson(fileName: string) {
+        const base = getBaseFileName(fileName);
+        return base === "package.json" || base === "bower.json";
+    }
+
+    function getDirectoryExcludingNodeModulesOrBowerComponents(f: string) {
+        const indexOfNodeModules = f.indexOf("/node_modules/");

```

Comment:
Don't we have helpers for this?  E.g. `commonPackageFolders` in core.ts.

Reply:
But TI uses only these two locations.